animation_nodes/nodes/vector/vector_noise.py

```diff
@@ -1,27 +1,277 @@
+    ("DISTANCE", "Distance", "", "", 0),
+    ("DISTANCE_SQUARED", "Distance Squared", "", "", 1),
+    ("MANHATTAN", "Manhattan", "", "", 2),
+    ("CHEBYCHEV", "Chebychev", "", "", 3),
+    ("MINKOVSKY", "Minkovsky", "", "", 4),
+    ("MINKOVSKY_HALF", "Minkovsky Half", "", "", 5),
+    ("MINKOVSKY_FOUR", "Minkovsky Four", "", "", 6)
+]
 
 class VectorNoiseNode(bpy.types.Node, AnimationNode, Noise3DNodeBase):
     bl_idname = "an_VectorNoiseNode"
     bl_label = "Vector Noise"
     bl_width_default = 160
 
+    noiseSelect = EnumProperty(name = "Type ", default = "FASTNOISE",
+        items = noiseSelectItems, update = AnimationNode.refresh)
+
+    blenderNoiseType = EnumProperty(name = "Type ", default = "NOISE",
+        description = "Blender noise type",
+        items = blenderNoiseTypeItems, update = AnimationNode.refresh)
+
+    noiseMode = EnumProperty(name = "Mode ", default = "FRACTAL",
+        description = "Blender noise mode",
+        items = noiseModeItems, update = AnimationNode.refresh)
+
+    noiseBasis = EnumProperty(name = "Basis ", default = "BLENDER",
+        description = "Noise basis type",
+        items = noiseBasisModeItems, update = AnimationNode.refresh)
+
+    noiseBasis2 = EnumProperty(name = "Basis ", default = "PERLIN_ORIGINAL",
+        description = "Noise basis type",
+        items = noiseBasisModeItems2, update = AnimationNode.refresh)
+
+    voronoiDistanceMetric = EnumProperty(name = "Mode ", default = "DISTANCE",
+        description = "Voronoi distance metric modes",
+        items = voronoiDistanceMetricItems, update = AnimationNode.refresh)        
+
+    useVectorList: VectorizedSocket.newProperty()                
+
     def create(self):
-        self.newInput("Vector List", "Vectors", "vectors")
-        self.createNoiseInputs()
-        self.newOutput("Float List", "Value", "value")
+        if self.noiseSelect == "FASTNOISE":
+            self.newInput("Vector List", "Vectors", "vectors")
+            self.createNoiseInputs()
+            self.newOutput("Float List", "Value", "value")
+        elif self.noiseSelect == "BLENDERNOISE":
+            if self.blenderNoiseType == "NOISE":
+                self.newInput(VectorizedSocket("Vector", "useVectorList",
+                    ("Vector", "vector"), ("Vectors", "vectors"))) 
+                self.newInput("Float", "Fractal Dimension", "fractalDimension")
+                self.newInput("Float", "Lacunarity", "lacunarity")
+                self.newInput("Integer", "Octaves", "octaves", value = 2, minValue = 1)
+
+                if self.noiseMode == "HETERO_TERRAIN":
+                    self.newInput("Float", "Offset", "offset")
+                if self.noiseMode in ["RIDGED_MULTI_FRACTAL", "HYBRID_MULTI_FRACTAL"]:
+                    self.newInput("Float", "Offset", "offset")
+                    self.newInput("Float", "Gain", "gain")
+
+                self.newOutput(VectorizedSocket("Float", ["useVectorList"],
```

Comment:
It is just a single value, no need of list`["useVectorList"] --> "useVectorList"`.
Same for other places.